clang/lib/DPCT/DPCT.cpp

```diff
@@ -740,8 +734,32 @@ int runDPCT(int argc, const char **argv) {
   CudaPath = getCudaInstallPath(OriginalArgc, argv);
   DpctDiags() << "Cuda Include Path found: " << CudaPath << "\n";
 
-  RefactoringTool Tool(OptParser->getCompilations(),
-                       OptParser->getSourcePathList());
+  ArrayRef<std::string> SourcePathList;
+  if (QueryAPIMapping.getNumOccurrences()) {
+    APIMapping::initEntryMap();
+    auto SourceCode = APIMapping::getAPISourceCode(QueryAPIMapping);
+    if (SourceCode.empty()) {
+      llvm::outs() << "The API Mapping is not available\n";
+      dpctExit(MigrationSucceeded);
+    }
+    int FD;
+    SmallString<64> TempFile;
+    if (auto EC =
+            llvm::sys::fs::createTemporaryFile("tmp", "cu", FD, TempFile)) {
+      llvm::errs() << "Error: dpct internal error. Cannot create temp file for "
```

Comment:
Maybe we need to add a new error code in `Error.h` WDYH?

Reply:
May be we should not create temp file later, so I prefer use this for now.